d2graph/d2graph.go

```diff
@@ -1549,7 +1549,11 @@ func (g *Graph) SetDimensions(mtexts []*d2target.MText, ruler *textmeasure.Ruler
 
 		obj.SizeToContent(contentBox.Width, contentBox.Height, paddingX, paddingY)
 	}
-	for _, edge := range g.Edges {
+	for i, edge := range g.Edges {
+		if edge.Link != nil {
+			g.Edges[i].Label.Value = edge.Link.Value
+		}
```

Comment:
this shouldn't be in this function named set dimensions. probably in compiler.go.

Also the link shouldn't replace the label if there is one. The test should show an example of where the link turns the label into a hyperlink